src/hooks/useLocalStorage.tsx

```diff
@@ -0,0 +1,37 @@
+import { useState, useCallback } from 'react';
+
+export const useLocalStorage = <T,>(key: string, initialValue: T) => {
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    if (typeof window === 'undefined') {
+      return initialValue;
+    }
+    try {
+      const item = window.localStorage.getItem(key);
+      return item ? JSON.parse(item) : initialValue;
+    } catch (error) {
+      console.log(error);
```

Comment:
why console.log is here ?

Reply:
Actually, this hook is became unused in the end of app developing, so I just will delete this whole file :) I used localStorage as a store to prevent redundant request to the server. It was as replacement of redux store.